### DISCUSSION TITLE
How to pass instruction discriminators

### DISCUSSION THREAD

**Discussion by 0xcen** [labeleven-dev/bettercallsol#102] [Q&A]
Anchor generates discriminators for Accounts as well as Instructions. How can I pass a discriminator as a byte array into a Better Call Sol transaction? I've tried this but it doesn't work. 
<img width="1027" alt="CleanShot 2023-03-31 at 11 26 54@2x" src="https://user-images.githubusercontent.com/75177443/229082062-1dfe50f2-f9e6-4c23-b9e2-d18ae100c02c.png">

**Comment by sohrab-** [comment]
If the program has an on-chain IDL or if you have an IDL JSON, then the instruction discriminator is taken care of for you. (Try "Anchor: Squads" in the "Examples" menu to see how that works.)

If you are trying to do it manually, currently we support utf8 strings for byte representation.

![image](https://user-images.githubusercontent.com/4444588/229401623-4dc5ecba-00a8-4684-9b84-77988a1e0a38.png)

This does give me an idea for an new enhancement: https://github.com/labeleven-dev/bettercallsol/issues/103
